### PR TITLE
Phase 2, logging refactor

### PR DESCRIPTION
**Pull request by Firstyear**
Implements 99, the second phase of the logging refactor. This adds tagged log attributes, and improves the format of what we collect so that it's easier to identify and trace specific areas of the server during development and administration.

- [ x ] cargo fmt has been run
- [ x ] cargo test has been run and passes
- [ - ] design document included (if relevant)

### CODE REVIEW

**Comment by Firstyear** [comment]
Also worth noting that due to these log changes, a few tweaks to indexing and query execution is in this PR, and it improves the performance of tests by 25% 😳

File: kanidmd/src/lib/audit.rs [hunk 1]
```diff
@@ -10,31 +10,172 @@ use chrono::DateTime;
...
+    Filter,
+    FilterError,
+    FilterWarning,
+    Trace,
+}
+
+impl fmt::Display for LogTag {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            LogTag::AdminError => write!(f, "admin::error 🚨"),
```
Comment (pando85) [inline kanidmd/src/lib/audit.rs:30]: I dunno if this icons will help or will make some difficulties showing logs and parsing them.

Comment (Firstyear) [reply kanidmd/src/lib/audit.rs:30]: Parsing there are still unique string tags associated, and I found the icons helpful? None of this is set in stone though, but for me when looking at the full log it helped things to standout. I'm attaching two screenshots as an example. Notice how easy it is to spot the access control lines? 

Screen Shot 2020-05-22 at 16 28 53

Screen Shot 2020-05-22 at 16 29 13

Comment (pando85) [reply kanidmd/src/lib/audit.rs:30]: I find them pretty cool, helps with visualization and with that format I think that they are easy to parse. So I love them.

**Pull request merged**